### PR TITLE
DBC-1315 - Fixing holdings.

### PR DESCRIPTION
**Pull request by bozeak**

### CODE REVIEW

File: includes/fbs.availability.inc [hunk 1]
```diff
@@ -154,8 +166,177 @@ function fbs_availability_holdings($provider_ids) {
...
+  // Set table rows.
+  $variables['rows'] = _fbs_set_rows($h);
+
+  // Set last row with reservation count.
+  $row = array();
+  $row['data']['Library'] = t('Reservations');
+  $row['data']['Copies'] = array('data' => $r_count, 'colspan' => 4);
+  $variables['rows'][] = $row;
+
+  return theme_table($variables);
```
Comment (avoykov) [inline includes/fbs.availability.inc]: pleae use theme('table', options)

File: includes/fbs.availability.inc [hunk 2]
```diff
@@ -154,8 +166,177 @@ function fbs_availability_holdings($provider_ids) {
...
+  }
+  return $rows;
+}
+
+/**
+ * Sum material for same placement in one row.
+ */
+function _sum_placement($placementsarr) {
+  $row = $placementsarr[0];
+  for ($i = 1; $i < count($placementsarr); $i++) {
```
Comment (avoykov) [inline includes/fbs.availability.inc]: why not foreach statement?

File: includes/fbs.availability.inc [hunk 3]
```diff
@@ -154,8 +166,177 @@ function fbs_availability_holdings($provider_ids) {
...
+  return $rows;
+}
+
+/**
+ * Collect materials with the same placement.
+ */
+function _get_placements_with_key($_rows, $currkey) {
+  $rows = array();
+  foreach ($_rows as $key) {
+    if ($key['placement'] == $currkey) {
```
Comment (avoykov) [inline includes/fbs.availability.inc]: you already looping needed placements, why you dont made actions from _sum_placement here?

File: includes/fbs.availability.inc [hunk 4]
```diff
@@ -154,8 +166,177 @@ function fbs_availability_holdings($provider_ids) {
...
+function _fbs_set_rows($h) {
+  $rows = array();
+
+  $copies_total = 0;
+  $home_total = 0;
+
+  foreach ($h as $key => $data) {
+    $row = array();
+
+    $row['placement'] = implode(' > ', $data['placement']);
```
Comment (avoykov) [inline includes/fbs.availability.inc]: $placement = implode(' > ', $data['placement']);
$copies = isset($data['total_count']) ? (int) $data['total_count'] : 0;
$home = isset($data['available_count']) ? (int) $data['available_count'] : 0;
$current = array(
  'copies' => isset($rows[$placement]['copies']) ? $rows[$placement]['copies'] : 0,
  'home' => isset($rows[$placement]['home']) ? $rows[$placement]['home'] : 0,
);

$rows[$placement] = array(
 'placement' => $placement,
 'copies' => $copies +$current['copies'] ,
 'home ' => $home  + $current['home'],
);

File: includes/fbs.availability.inc (hunk 2 above)
Comment (bozeak) [inline includes/fbs.availability.inc]: IMHO "for" statement is more suitable for this increment operation.

File: includes/fbs.availability.inc (hunk 3 above)
Comment (bozeak) [inline includes/fbs.availability.inc]: Didn't wanted to mess all things in one place.

**Pull request merged**